pages/scripts/exporthardcopy.js

```diff
@@ -0,0 +1,271 @@
+                    '<tr><td class="lh_jhc_caller"><strong>Caller:</strong> ++CallerFirstName++ ++CallerLastName++ ++CallerPhoneNumber++</td></tr>'+
+                    '<tr><td class="lh_jhc_contact"><strong>Job Contact:</strong> ++ContactFirstName++ ++ContactLastName++ ++ContactPhoneNumber++</td></tr>'+
+                    '<tr><td class="lh_jhc_access"><strong>Property Access:</strong> ++PermissionToEnterPremises++++howToEnterPremises++</td></tr>'+
+                    '<tr><td height="30">&nbsp;</td></tr>'+
+                    '<tr><th class="lh_jhc_attend_head">Attendance Details</th></tr>'+
+                    '<tr class="lh_jhc_form"><td>Team or Team Leader Name:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>Date &amp; Time Of Arrival:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>Date &amp; Time Of Departure:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>Description of Situation:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>Work Performed:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>Any Further Work Needed:</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                    '<tr class="lh_jhc_form"><td>&nbsp;</td></tr>'+
+                      /*
+                    '<tr>'+
+                      '<td>'+
+                        '<table class="jobHardcopy_tasker">'+
+                          '<tr><td>++tasker++<br/>Email: ++tasker_email++<br/>Phone: ++tasker_phone++<br/>Fax: ++tasker_fax++</td></tr>'+
+                          '<tr><td></td></tr>'+
+                        '</table>'+
+                      '</td>'+
+                    '</tr>'+
+                      */
+                  '</table>'+
+                  '<hr/>';
+
+//make the call to beacon
+function HackTheMatrix(id, host, progressBar) {
+  console.log('HackTheMatrix( %s , %s , %o )',id, host, progressBar);
+  var unit = [];
+  var start = new Date(decodeURIComponent(params.start));
+  var end = new Date(decodeURIComponent(params.end));
+
+  if(selectedcolumns.length == 0){
+    console.log('No Fields Selected');
+    return false;
+  }else{
+    console.log('selectedcolumns = %o' , selectedcolumns);
+  }
+
+  if (typeof id !== 'undefined') {
+    id.split(',').forEach(function(d){
+      var newObj = {'Id': d};
+      unit.push(newObj);
+    });
+  }
+
+  if(unit.length == 0){
+    console.log('No Units Selected');
+  }else{
+    console.log(unit);
+  }
+
+
+  LighthouseJob.get_json(unit, host, start, end, function(jobs) {
+
+    $.each(jobs.Results , function(k,v){
```

Comment:
`$.each` and `_.each` one or the other please? imo prefer `_` since it's designed for working with collections.